src/Advection/centered_fourth_order.jl

```diff
@@ -3,11 +3,16 @@
 #####
 
 """
-    struct CenteredFourthOrder <: AbstractCenteredAdvectionScheme{1}
+    struct CenteredFourthOrder <: AbstractCenteredAdvectionScheme{2}
 
 Centered fourth-order advection scheme.
 """
-struct CenteredFourthOrder <: AbstractCenteredAdvectionScheme{1} end
+struct CenteredFourthOrder{CA} <: AbstractCenteredAdvectionScheme{2} 
+    "advection scheme used near boundaries"
+    child_advection :: CA
```

Comment:
`boundary_scheme` or `near_boundary_scheme`?

We call it "advection" in the context of a model, but here, the type is in the context of another advection scheme.